nixos/modules/security/systemd-harden.nix

```diff
@@ -0,0 +1,115 @@
+
+{config, lib, ...}: let
+  inherit (lib) types;
+in {
+  options.systemd.services = lib.mkOption {
+    type = types.attrsOf (types.submodule ({
+      name,
+      config,
+      ...
+    }: {
+      options.harden = {
+        enable = lib.mkOption {
+          type = types.bool;
+          default = false;
+          description = lib.mdDoc ''
+            Basic restrictions for systemd services
+          '';
+        };
+
+        execOnlyNix = lib.mkOption {
+          type = types.bool;
+          default = true;
+          description = lib.mdDoc ''
+            Mark whole system as non-executable with exception for `/nix/store`.
+          '';
+        };
```

Comment:
The problem with `mkEnableOption` is that it has very limited support for docs. Current docs are more of a stub rather than target docs and I would like to expand them more when there will be established syntax for all these options.

Reply:
I don't see how `mkEnableOption` would limit your docs in any way? The description is its one and only argument, it's just prefixed by "Whether to enable ..." which applies to all of them.